homeassistant/components/open_meteo/weather.py

```diff
@@ -55,6 +60,15 @@ def __init__(
             name=entry.title,
         )
 
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        super()._handle_coordinator_update()
+        assert self.platform.config_entry
```

Comment:
Let's get rid of the assert.

Reply:
The assert is there because the `platform` does not always have a `config_entry`, the same is true for `DataUpdateCoordinator`

Reply:
Fixed, `OpenMeteoWeatherEntity` now extends `CoordinatorWeatherEntity`